cluster.go

```diff
@@ -304,17 +304,22 @@ func (clstr *Cluster) tend() Error {
 				logger.Logger.Warn("Peer node `%s` is different than actual node `%s` for host `%s`", _peer.nodeName, nv.name, host)
 			}
 
-			if clstr.peerExists(peers, nv.name) {
-				// Node already exists. Do not even try to connect to hosts.
-				return seq.Break
-			}
-
 			// Create new node.
 			node := clstr.createNode(&nv)
 			peers.addNode(nv.name, node)
 			partMap.InitDoVal(clstr.getPartitions().clone, func(partMap partitionMap) {
 				node.refreshPartitions(peers, partMap, true)
 			})
+
+			if _peer.replaceNode != nil {
+				// Preventing duplicate entries.
+				for _, entry := range peers._nodesToRemove {
+					if entry.Equals(_peer.replaceNode) {
+						return seq.Break
+					}
+				}
+				peers._nodesToRemove = append(peers._nodesToRemove, _peer.replaceNode)
```

Comment:
`seq.ParDo` does stuff concurrently for all the list elements, so there is a potential race condition here. We probably need a `addNodesToRemove()` method on `peers` struct, and `peers._nodesToRemove` should be moved into a `SyncVal` or something. The above logic can also be moved into that method, simplifying the code here.